includes/Admin/Admin_AJAX.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Admin\Admin_AJAX
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Admin;
+
+use WP_Error;
+
+/**
+ * Class to handle the Admin AJAX requests.
+ *
+ * @since n.e.x.t
+ */
+class Admin_AJAX {
+
+	/**
+	 * Nonce key.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const NONCE_KEY = 'plugin-check-run-checks';
+
+	/**
+	 * Registers WordPress hooks for the Admin AJAX.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function add_hooks() {
+		add_action( 'wp_ajax_plugin_check_run_checks', array( $this, 'run_checks' ) );
+	}
+
+	/**
+	 * Creates and returns the nonce.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function get_nonce() {
+		return wp_create_nonce( self::NONCE_KEY );
+	}
+
+	/**
+	 * Run checks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function run_checks() {
+		$nonce = filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_STRING );
+
+		if ( ! wp_verify_nonce( $nonce, self::NONCE_KEY ) ) {
+			wp_send_json_error(
+				new WP_Error( 403, __( 'Invalid nonce', 'plugin-check' ) ),
```

Comment:
The error code should not be an HTTP status code, but a slug like string that identifies the error.
```suggestion
				new WP_Error( 'invalid-nonce', __( 'Invalid nonce', 'plugin-check' ) ),
```